### PR TITLE
generate_enc_password: increase rsalt by 2

### PR DESCRIPTION
**Pull request by steveej**
before this change i see a bufferflow during `readable_string`.

### CODE REVIEW

**Comment by ggarber** [comment]
Thank you very much for the report @steveej and the fix! 🙇  
I think the problem is in the readable_string function and we shouldn't workaround it here, but let me take a look.

**Comment by ggarber** [comment]
@steveej Could you try this alternative fix?   Thank you!

static void readable_string(unsigned char *orig, unsigned char *out, size_t sz) {
  size_t i = 0;
  out[0] = 0;

  for (i = 0; i < sz; ++i) {
    snprintf((char *)(out + (i * 2)), 3, "%02x", (unsigned int)orig[i]);
  }
  out[sz * 2] = 0;
}

File: src/client/ns_turn_msg_defs.h [hunk 1]
```diff
@@ -60,9 +60,9 @@
 #define GET_STUN_ERR_RESP(msg_type) (msg_type | 0x0110)
 
 /* Lifetimes: */
-#define STUN_DEFAULT_ALLOCATE_LIFETIME (600)
+#define STUN_DEFAULT_ALLOCATE_LIFETIME (20)
```
Comment (eakraly) [inline src/client/ns_turn_msg_defs.h:63]: Were these changed for debugging?
Those should be reverted

**Review by eakraly — CHANGES_REQUESTED** [review]

**Comment by ggarber** [comment]
@steveej Can you take a look at the comments in this PR when you have a minute? Thank you very much !

File: src/client/ns_turn_msg_defs.h (hunk 1 above)
Comment (steveej) [reply src/client/ns_turn_msg_defs.h]: thanks for catching this, those commits weren't supposed to be in this PR. i took them out however i still have other suggestions to consider.

**Comment by steveej** [comment]
hey @ggarber all comments have been addressed

**Comment by ggarber** [comment]
Thank you very much @steveej !

**Pull request merged**